src/lsm/compaction.zig

```diff
@@ -463,6 +476,69 @@ pub fn CompactionType(
             }
         }
 
+        // TODO: We need to reimplement the .secondary_index optimization for sort here.
```

Comment:
What optimization is this referring to? The tombstone-drop optimization?

Reply:
Yeah - I'll elaborate in the comment. For general tables, multiple tree `remove` / `put` operations are consolidated, but for secondary indexes they're not because the tombstone has a different `timestamp` value in its composite key, so if we're a secondary index we should be able to take the latest just by `field`.

Reply:
... actually I might just implement it quickly :smile:

Reply:
@sentientwaffle famous last words... It's a bit more complex than I thought initially, so I've expanded the comment and leaving it for a follow-up PR.